### ISSUE TITLE
One vs All Classification

### ISSUE THREAD

**Issue by CarlosUziel** [apache/incubator-mxnet#4029]
Hi,

I'm using a CNN in R to perform image classification with K different classes. However, I want to try another approach and do a "one vs all" classification with each label, so one label would be 1 and the rest would be 0.

My question is, what output layer should I use? If I use mx.symbol.SoftmaxOutput after a FullyConnected layer with one neuron is useless because it always predict 1 or even NaN for every example. LinearRegressionOutput gives a similar result.

My basic idea is that I need an output layer that performs 0.5 threshold on the single output neuron from the last FullyConnected layer.

Sorry if this is a dumb question, I'm still new to this field.
Thanks in advanced.

**Comment by piiswrong** [comment]
LogisticRegressionOutput

**Comment by CarlosUziel** [comment]
@piiswrong Exactly what I was looking for, it worked like a charm. Thank you very much!

**Comment by yuan1202** [comment]
hi @piiswrong 

Thanks for the answer, exactly what i am looking for.

Just want to add a nab question following this issue:
- which metric would be best suited for this kind of 1 vs all classification problem.

I suppose standard accuracy is not good because of the big difference in the number of samples from the two classes.

**Comment by CarlosUziel** [comment]
Hi @yuan1202 ,

I think I may have an answer for you. During the training phase, I just use the rmse metric, but for further analysis I do the following:

First, I make a prediction using the test set, and afterwards I analyse all the different metrics you can obtain from the confusion matrix (as you can see [here](https://en.wikipedia.org/wiki/Receiver_operating_characteristic)). Also, you may be interested in the F1 Score metric.

I hope this is what you were looking for.

**Comment by yuan1202** [comment]
Hey @CarlosUziel 

Thank you for the reply.

I will try out your approach.

I have a rather biased binary class situation where 0 class to 1 class ratio is about 19:1. But I suppose it's similar for all one vs all situation.

When I just use default accuracy metric the result is around 95% which seems not right to me because it's too close to the 19:1 ratio.

I tried to use F1 score but it's constantly stuck at 0.000000 no matter the number of epoch or output layer being SoftmaxOutput or LogisticRegressionOutput. So I am no sure where I got it wrong.
